File: src/typescript/Calculate average.ts
/* 
Write a function which calculates the average of the numbers in a given array.

Note: Empty arrays should return 0.
*/

export function findAverage(array: number[]): number {
  const sum = array.reduce((acc, curr) => acc + curr, 0);
  return sum / array.length || 0;
}

File: src/lua/Calculate average.lua
-- Write a function which calculates the average of the numbers in a given array.

-- Note: Empty arrays should return 0.
local function find_average(numbers)
	local total = 0
	local length = #numbers
	for i = 1, length do
		total = total + numbers[i]
	end
	return length > 0 and (total / length) or 0
end

return find_average
